### PR TITLE
Update Content Management UI behavior

### LINKED ISSUE #2207: 'transfer failed' state has no way to get out of it
The 'transfer failed' state has no mechanism for proceeding - the user is stuck here:

image

cc @jonboiser

### PR DESCRIPTION
**Pull request by jonboiser**
This PR

1. For #2237 Adds more information to "finished" message on progress bar that tells people to press "close" to see updates to the channel list. In the case of imports, however, the user will (prematurely) see the newly-imported channel if user lands on page after import has begun. Similarly, deleted channels will not appear when user lands on page after it's deleted.
1. Removes the use of `uiAlert` on page and just uses messages in progress bar to provide feedback.
1. Fixes #2207 by fixing logic for showing cancel button.
1. Removes unnecessary mutation that updates the core channel list on every task poll.
1. Disables the 'delete' buttons on the channel list whenever there is some Task in progress.
1. Adds some logic to make sure different browsers are better synchronized

UPDATES:
1. ~For #1927, incorporated hack to approximate "cumulative progress" for multi-part transfers (for local imports only. export and remote import are always one part for now). Test by importing from USB with multiple channels.~
1. ~For #2206 Incorporate hack to adjust for when Task % spikes to 100% immediately after it starts. (Seems to happen more reliably with remote import than local, so try that scenario).~

**Cancelling task when deletion fails**

delete fail 2

**New finish message**

close message

### CODE REVIEW

**Comment by indirectlylit** [comment]
This is a big improvement. There are still some subtle, weird behaviors around when the channel list updates (see below). However we can tackle that in 0.7, maybe with some input from Jessica and Khang.

This change introduces minor string changes (cc @rtibbles @carinediaz) but I think is small enough to squeeze in (or leave in english, if absolutely necessary)


gifmakerproject25

File: kolibri/plugins/management/assets/src/device_management/views/manage-content-page/index.vue [hunk 1]
```diff
@@ -9,12 +9,9 @@
         <task-progress
           v-if="tasksInQueue"
           v-bind="firstTask"
-          @taskcomplete="showTaskCompleteNotification()"
-          @taskfailed="showTaskFailedNotification()"
+          @cleartask="clearFirstTask"
         />
 
-        <notifications v-bind="{notification}" @dismiss="clearNotification()" />
```
Comment (indirectlylit) [inline kolibri/plugins/management/assets/src/device_management/views/manage-content-page/index.vue:16]: nice

**Comment by jonboiser** [comment]
Handling the weird behavior (prematurely-available channels) will probably need to wait until the next version because the APIs to manage this aren't available yet.

Some fixes to consider would be

1. for Kolibri to "clean up" partial downloads after a cancel, and delete the database and contents.
1. defer putting the channel into the ContentMetaDB resource until all of its resources are downloaded.

**Comment by rtibbles** [comment]
Not sure when prematurely available channels occur? They will only appear to learners after the annotation step happens, which is after content download has completed.

**Comment by jonboiser** [comment]
It seems correct that the channel is not immediately available to learners in mid-download. But on the Manage Content Page, if you refresh mid-download (or visit the page on a different browser mid-download), then the Channel will appear on the list, as if it were 100% complete.

screen shot 2017-10-02 at 4 43 29 pm

**Comment by rtibbles** [comment]
One quick fix would be to do the resource number annotation and file size only by available content, so at least it would say 0, 0Mb.

**Pull request merged**